### ISSUE TITLE
can't find inner product in semantic learning branch from code "att_normalization"

### ISSUE THREAD

**Issue by RFK97** [shepnerd/AttenNorm#6]
The paper says "We give n initial desired semantic entities, and define their correlation to the feature points of the image **as their inner product**" 

but in code, there's no inner product of any feature points.

_x_mask_filters = tf.Variable(tf.random_normal(shape=[1, 1, c, nClass]), name='mask_filters')
x_mask = tf.nn.conv2d(x, x_mask_filters, strides=[1,1,1,1], padding='VALID')
layout = tf.nn.softmax((alpha * fast_activations + x_mask) / kama, axis=3) # b x h x w x n_

plus, in the Regional Normalization part of the code, xv is used instead of x (which is written in paper).

Could you understand me? thank you.

**Comment by shepnerd** [comment]
In this code snipplet, `x_mask_filters` stands for n semantic entities, and we conduct the inner product between these entities and the image feature points using convolution by treating n entities as convolutional filters, just as
`x_mask = tf.nn.conv2d(x, x_mask_filters, strides=[1,1,1,1], padding='VALID')`
This is also described in our paper L20-26, P3.

**Comment by RFK97** [comment]
I understood. Thank you.

**Comment by RFK97** [comment]
Also, could you let me know where's denormalization part in code? 
_xn = tf.reduce_sum((xv_expand-xv_mean)/(xv_std+eps)*layout_expand, axis=-1)_
seems to do Regional Normalization without Denormalization.
Sorry for bothering. Thank you!

**Comment by shepnerd** [comment]
For generative inpainting, we do not apply denormalization to the regional normalization.
